app/knight.py

```diff
@@ -0,0 +1,40 @@
+class Knight:
+    protection = 0
+
+    def __init__(
+        self,
+        name: str,
+        power: int,
+        hp: int,
+        armour: list,
+        weapon: dict,
+        potion: dict
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+
+    def set_armour(self) -> None:
+        if self.armour:
+            for equipment in self.armour:
+                self.protection += equipment.get("protection")
+
+    def set_weapon(self) -> None:
+        self.power += self.weapon.get("power")
+
+    def set_potion(self) -> None:
+        if self.potion:
+            if "power" in self.potion.get("effect"):

```

Comment:
You do not need to call `self.potion.get("effect")` three times throughout these if conditions, you can call it once and write it to a separate variable.